python/ql/lib/semmle/python/regex.qll

```diff
@@ -1,1105 +1,43 @@
-    )
-  }
-
-  /**
-   * Whether the text in the range start,end is an alternation
-   */
-  predicate alternation(int start, int end) {
-    this.top_level(start, end) and
-    exists(int less | this.subalternation(start, less, _) and less < end)
-  }
-
-  /**
-   * Whether the text in the range start,end is an alternation and the text in part_start, part_end is one of the
-   * options in that alternation.
-   */
-  predicate alternationOption(int start, int end, int part_start, int part_end) {
-    this.alternation(start, end) and
-    this.subalternation(start, part_end, part_start)
-  }
-
-  /** A part of the regex that may match the start of the string. */
-  private predicate firstPart(int start, int end) {
-    start = 0 and end = this.getText().length()
-    or
-    exists(int x | this.firstPart(x, end) |
-      this.emptyMatchAtStartGroup(x, start) or
-      this.qualifiedItem(x, start, true, _) or
-      // ^ and \A match the start of the string
-      this.specialCharacter(x, start, ["^", "\\A"])
-    )
-    or
-    exists(int y | this.firstPart(start, y) |
-      this.item(start, end)
-      or
-      this.qualifiedPart(start, end, y, _, _)
-    )
-    or
-    exists(int x, int y | this.firstPart(x, y) |
-      this.groupContents(x, y, start, end)
-      or
-      this.alternationOption(x, y, start, end)
-    )
-  }
-
-  /** A part of the regex that may match the end of the string. */
-  private predicate lastPart(int start, int end) {
-    start = 0 and end = this.getText().length()
-    or
-    exists(int y | this.lastPart(start, y) |
-      this.emptyMatchAtEndGroup(end, y)
-      or
-      this.qualifiedItem(end, y, true, _)
-      or
-      // $ and \Z match the end of the string.
-      this.specialCharacter(end, y, ["$", "\\Z"])
-    )
-    or
-    this.lastPart(_, end) and
-    this.item(start, end)
-    or
-    exists(int y | this.lastPart(start, y) | this.qualifiedPart(start, end, y, _, _))
-    or
-    exists(int x, int y | this.lastPart(x, y) |
-      this.groupContents(x, y, start, end)
-      or
-      this.alternationOption(x, y, start, end)
-    )
-  }
-
-  /**
-   * Whether the item at [start, end) is one of the first items
-   * to be matched.
-   */
-  predicate firstItem(int start, int end) {
-    (
-      this.characterItem(start, end)
-      or
-      this.qualifiedItem(start, end, _, _)
-      or
-      this.charSet(start, end)
-    ) and
-    this.firstPart(start, end)
-  }
-
-  /**
-   * Whether the item at [start, end) is one of the last items
-   * to be matched.
-   */
-  predicate lastItem(int start, int end) {
-    (
-      this.characterItem(start, end)
-      or
-      this.qualifiedItem(start, end, _, _)
-      or
-      this.charSet(start, end)
-    ) and
-    this.lastPart(start, end)
+class StdLibRegExpInterpretation extends RegExpInterpretation::Range {
+  StdLibRegExpInterpretation() {
+    this =
+      API::moduleImport("re").getMember(any(string name | name != "escape")).getACall().getArg(0)
```

Comment:
I was about to suggest we remove this class altogether, because `RegExpInterpretation` is always used together with `any(Concepts::RegexExecution exec).getRegex()`, but I can see that if a compiled regex is never executed , it will not become a `RegexExecution` :thinking: (this can also happen if our tracking of it fails to determine it reaches an execution)

So how about the following?

```suggestion
    this =
      API::moduleImport("re").getMember("compile").getACall().getParameter(0, "pattern").asSink()
```

Also, could we move this into `Stdlib.qll` next to the other regex modeling?